### ISSUE TITLE
Issue recreating connection to a queue after a connection problem

### ISSUE THREAD

**Issue by dd185209** [mosquito/aio-pika#632]
Hi my application is using aio-pika=9.4.1 to connect to a rabbitmq. I'm testing under docker environment and if I manually shutdown rabbit I see following log messages:
```
Unexpected connection close from remote "amqp://user:******@localhost:5672", Connection.Close(reply_code=320, reply_text="CONNECTION_FORCED - broker forced connection closure with reason 'shutdown'")
NoneType: None
Connection attempt to "amqp://user:******@localhost" failed: Server connection unexpectedly closed. Read 0 bytes but 1 bytes expected. Reconnecting after 5 seconds.
Connection attempt to "amqp://user:******@localhost" failed: [WinError 1225] The remote computer refused the network connection. Reconnecting after 5 seconds.
Connection attempt to "amqp://user:******@localhost" failed: [WinError 1225] The remote computer refused the network connection. Reconnecting after 5 seconds.
Connection attempt to "amqp://user:******@localhost" failed: [WinError 1225] The remote computer refused the network connection. Reconnecting after 5 seconds.
```
but when rabbit restart I see following errors: 
```
Connection attempt to "amqp://user:******@localhost" failed: Server connection unexpectedly closed. Read 0 bytes but 1 bytes expected. Reconnecting after 5 seconds.
Connection attempt to "amqp://user:******@localhost" failed: Server connection unexpectedly closed. Read 0 bytes but 1 bytes expected. Reconnecting after 5 seconds.
```
then the application is successfully connecting to new instance but not to the exchange and queue(there is no consumer).
and the application is blocked whiting for new messages that never come. 

Using aio-pika==8.2.1 the application were working fine, I got similar error when the rabbitmq is down, but when it comes up again even the consumer is connected and messages are processed correctly.

**Comment by Darsstar** [comment]
This could be a duplicate of https://github.com/mosquito/aio-pika/issues/594.

How certain are you that there is a strong reference chain to the Channel or Queue?

**Comment by drsnyder** [comment]
Also confirmed that 9.4.1 does not reconnect after a rabbit restart but 8.2.1 does. In my testing, the application gets blocked on [this line](https://github.com/mosquito/aio-pika/blob/7e468aa229cb751430a31b5ba00d42f1e1373cd0/aio_pika/robust_connection.py#L124) waiting for the `__connection_close_event` to be set.

**Comment by mosquito** [comment]
@drsnyder did you try to upgrade to the latest release?

**Comment by drsnyder** [comment]
Yes I tried upgrading. Release [9.4.2](https://github.com/mosquito/aio-pika/releases/tag/9.4.2) also does not work.

**Comment by mosquito** [comment]
Current version is https://pypi.org/project/aio-pika/9.5.4/

**Comment by drsnyder** [comment]
@mosquito, thanks for getting back to me so fast. Sorry, 9.5.4 doesn't work either. It appears to be reconnecting but the consumer call backs are not receiving messages. Similar to the OP, I get:

```
Unexpected connection close from remote "amqp://<user>:******@localhost:5672//?reconnect_interval=10&fail_fast=0", Connection.Close(reply_code=320, reply_text="CONNECTION_FORCED - broker forced connection closure with reason 'shutdown'")
NoneType: None
Connection attempt to "amqp://<user>:******@localhost:5672//?reconnect_interval=10&fail_fast=0" failed: Server connection unexpectedly closed. Read 0 bytes but 1 bytes expected. Reconnecting after 10 seconds.
Connection attempt to "amqp://<user>:******@localhost:5672//?reconnect_interval=10&fail_fast=0" failed: Server connection unexpectedly closed. Read 0 bytes but 1 bytes expected. Reconnecting after 10 seconds.
Connection attempt to "amqp://<user>:******@localhost:5672//?reconnect_interval=10&fail_fast=0" failed: Server connection unexpectedly closed. Read 0 bytes but 1 bytes expected. Reconnecting after 10 seconds.
Connection attempt to "amqp://<user>:******@localhost:5672//?reconnect_interval=10&fail_fast=0" failed: Server connection unexpectedly closed. Read 0 bytes but 1 bytes expected. Reconnecting after 10 seconds.
```

**Comment by paul-godhouse** [comment]
hi @drsnyder, did you manage to sort it out?

I'm in a similar situation. The connection between my worker and rabbitmq is lost for some (legitimate) reason. 
In the end, the queues remain but the consumers are gone. Therefore the messages are not acknowledged anymore. 

The only solution I found for now is to kill my pod on k8s to restart the connection from scratch. This way, the consumers are properly recreated

N.B: I don't use aio-pika directly, I'm using faststream
